src/patternfly/components/Page/page.scss

```diff
@@ -482,12 +480,21 @@
     padding: 0;
   }
 
-  &.pf-m-no-padding-mobile {
-    @media screen and (max-width: $pf-global--breakpoint--md) {
-      --pf-c-page__main-section--PaddingTop: var(--pf-c-page__main-section--m-no-padding-mobile--md--PaddingTop);
-      --pf-c-page__main-section--PaddingRight: var(--pf-c-page__main-section--m-no-padding-mobile--md--PaddingRight);
-      --pf-c-page__main-section--PaddingBottom: var(--pf-c-page__main-section--m-no-padding-mobile--md--PaddingBottom);
-      --pf-c-page__main-section--PaddingLeft: var(--pf-c-page__main-section--m-no-padding-mobile--md--PaddingLeft);
+  &.pf-m-no-padding-on-sm,
+  &.pf-m-no-padding-on-md,
+  &.pf-m-no-padding-on-lg,
+  &.pf-m-no-padding-on-xl,
+  &.pf-m-no-padding-on-2xl {
+    padding: 0;
+  }
+
+  @each $breakpoint, $breakpoint-value in $pf-c-page--breakpoint-map {
+    $breakpoint-name: if($breakpoint != "base", -on-#{$breakpoint}, "");
+
+    @include pf-apply-breakpoint($breakpoint, $pf-c-page--breakpoint-map) {
+      &.pf-m-no-padding#{$breakpoint-name} {
+        padding: var(--pf-c-page__main-section--PaddingTop) var(--pf-c-page__main-section--PaddingRight) var(--pf-c-page__main-section--PaddingBottom) var(--pf-c-page__main-section--PaddingLeft);
```

Comment:
```suggestion
                --pf-c-page__main-section--PaddingTop: 0;
                --pf-c-page__main-section--PaddingRight: 0;
                --pf-c-page__main-section--PaddingBottom: 0;
                --pf-c-page__main-section--PaddingLeft: 0;
```

Reply:
were you able to test this? because`pf-apply-breakpoint` is set to min-width, it removes the padding at `md` and above, when we want the reverse. That's why I had removed the padding from the selectors and then added it back in at the breakpoint that is specified. but maybe im missing something! let me know what you think

Reply:
Oh I see... personally I think that's confusing, since all of our other breakpoint modifiers apply the modifier name at the specified breakpoint and above, with exceptions like the `pf-m-grid` modifiers on table. wdyt @mattnolting